spec/features/professors/index_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe "Index" do
+  before :each do
+    @snape = Professor.create(name: "Severus Snape", age: 45, specialty: "Potions")
+    @hagarid = Professor.create(name: "Rubeus Hagrid", age: 38 , specialty: "Care of Magical Creatures")
+    @lupin = Professor.create(name: "Remus Lupin", age: 49 , specialty: "Defense Against The Dark Arts")
+
+    @harry = Student.create!(name: "Harry Potter" , age: 11 , house: "Gryffindor" )
+    @malfoy = Student.create!(name: "Draco Malfoy" , age: 12 , house: "Slytherin" )
+    @longbottom = Student.create!(name: "Neville Longbottom" , age: 11 , house: "Gryffindor" )
+
+  end
+
+  it 'can show a list of professors along with their name, age, and specialty' do
+    visit '/professors'
+
+    expect(page).to have_content(@snape.name)
+    expect(page).to have_content(@snape.age)
+    expect(page).to have_content(@snape.specialty)
+
+    expect(page).to have_content(@hagarid.name)
+    expect(page).to have_content(@hagarid.age)
+    expect(page).to have_content(@hagarid.specialty)
+
+    expect(page).to have_content(@lupin.name)
+    expect(page).to have_content(@lupin.age)
+    expect(page).to have_content(@lupin.specialty)
+
+  end
+
+  it 'lists professors in alphabetical order' do
+     visit '/professors'
+
+     expect(@lupin.name).to appear_before(@hagarid.name)
+     expect(@hagarid.name).to appear_before(@snape.name)
+   end
+
```

Comment:
The orderly gem is probably the most succinct way to check for order. I agree it doesn't seem very efficient. You might consider doing something like this:

Assuming that you are wrapping the data you are displaying in divs in the view, your view might look something like this:
```
@people.each do |person|
  <div class="name"> person.name </div>
end
```

In your test you can do something like:
```
expected_order = [@person1.name, @person3.name, "@person2.name] // ["Alex", "Ben", "Candice"]
actual_order = page.all(".name").map do |name|
  name.text
end
expect(actual_order).to eq(expected_order)
```
This is utilizing the [page.all ](https://rubydoc.info/github/jnicklas/capybara/master/Capybara%2FNode%2FFinders:all) capybara method to map over all of the '.name' divs on the page and return the text from each div, then put that text into an array.